File: recursions/find_factorial/find_factorial.c
#include <stdio.h>

int findFactorialIterative(int num);
int findFactorialRecursion(int num);

int main(int argc, char *argv[]) {

    int iterative = findFactorialIterative(5);
    printf("Iterative function: %d\n", iterative);

    int recursive = findFactorialRecursion(6);
    printf("Recursion function: %d\n", recursive);

    return 0;
}

int findFactorialIterative(int num) {
    int answer = 1;
    if (num == 0) {
        return answer;
    }

    for (int i = 1; i <= num; ++i) {
        answer = answer * i;
    }
    
    return answer;
}

int findFactorialRecursion(int num) {
    if (num == 0 || num == 1) {
        return 1;
    } else if (num == 2) {
        return 2;
    }

    return num * findFactorialRecursion(num-1);
}

File: recursions/find_factorial/find_factorial.php
<?php

function findFactorialRecursive($num) {
    if ($num === 0 || $num === 1) {
        return 1;
    } else if ($num === 2) {
        return 2;
    }

    return $num * findFactorialRecursive($num-1);
}

$recursive = findFactorialRecursive(5);
echo "Recursive function: " . $recursive . "\n";

function findFactorialIterative($num) {
    $counter = 0;
    $answer = 1;
    if ($num === 0) {
        return $answer;
    }

    for ($i = 0; $i < $num; ++$i) {
        $counter++;
        $answer = $answer * $counter;
    }

    return $answer;
}

$iterative = findFactorialIterative(5);
echo "Itertative function: " . $iterative . "\n";
